component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1197,6 +1199,11 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                 Object verifiedMobileObject = context.getProperty(SMSOTPConstants.REQUESTED_USER_MOBILE);
                 if (verifiedMobileObject != null) {
                     try {
+                        if (Utils.getThreadLocalToSkipSendingSmsOtpVerificationOnUpdate() != null) {
```

Comment:
If we are setting the value immediately next, no need to unset this.

Reply:
Addressed with https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/184/commits/a8f6499b65634323b72a5dddef521d0c14c91b6a